service-discovery/stork-consul/src/test/java/io/quarkus/ts/stork/StorkServiceDiscoveryIT.java

```diff
@@ -0,0 +1,163 @@
+        String response = makePingCall("pung");
+        assertThat("Service discovery by name fail.", PREFIX + "pung", is(response));
+    }
+
+    @Test
+    @Order(2)
+    public void storkLoadBalancerByRoundRobin() {
+        Map<String, Integer> uniqueResp = new HashMap<>();
+        final int requestAmount = 100;
+        final int roundRobinError = (requestAmount / 2) - 1;
+        for (int i = 0; i < requestAmount; i++) {
+            String response = makePingCall("pong");
+            if (uniqueResp.containsKey(response)) {
+                uniqueResp.put(response, uniqueResp.get(response) + 1);
+            } else {
+                uniqueResp.put(response, 1);
+            }
+        }
+
+        Assertions.assertEquals(uniqueResp.size(), 2, "Only 2 services should response");
+        assertThat("Unexpected service names", uniqueResp.keySet(),
+                hasItems(PREFIX + DEFAULT_PONG_RESPONSE, PREFIX + DEFAULT_PONG_REPLICA_RESPONSE));
+        assertThat("Load balancer doesn't behaves as round-robin", uniqueResp.get(PREFIX + DEFAULT_PONG_RESPONSE),
+                is(greaterThanOrEqualTo(roundRobinError)));
+        assertThat("Load balancer doesn't behaves as round-robin", uniqueResp.get(PREFIX + DEFAULT_PONG_REPLICA_RESPONSE),
+                is(greaterThanOrEqualTo(roundRobinError)));
+    }
+
+    @Test
+    @Order(3) // run this test always last, because we are dropping pongService
+    @Disabled("Looks that service-discovery.refresh-period is not aware about service readiness")
+    //TODO https://github.com/quarkusio/quarkus/issues/24343
+    public void storkLoadBalancerServiceNodeDown() {
+        final int requestAmount = 100;
+        final int nodeDownThreshold = 50;
+        final Duration nodeDownRecoveryTimeout = Duration.ofSeconds(10);
+        for (int i = 0; i < requestAmount; i++) {
+            Awaitility.await().atMost(nodeDownRecoveryTimeout).untilAsserted(() -> makePingCall("pong"));
+            if (i > nodeDownThreshold) {
+                pongService.stop();
+            }
+        }
+
+        String respBody = makePingCall("pong");
+        Assertions.assertEquals(respBody, PREFIX + DEFAULT_PONG_REPLICA_RESPONSE, "Unexpected service name");
+    }
+
+    private String makePingCall(String subPath) {
+        return mainPingService
+                .given()
+                .get("/ping/" + subPath).then()
+                .statusCode(HttpStatus.SC_OK)
+                .extract().body().asString();
+    }
+
+    private static String getConsultEndpoint(String endpoint) {
+        return endpoint.replaceFirst(":\\d+", "");
+    }
+
+    private static String getAvailablePort() {
+        int port = 0;
+        try (ServerSocket serverSocket = new ServerSocket(port)) {
+            return String.valueOf(serverSocket.getLocalPort());
+        } catch (IOException e) {
+            fail("no free port available.");
+        }
+
+        return String.valueOf(port);
```

Comment:
We do not need this return statement(and, probably, the "port" variable)

Reply:
This is why was a better idea to leave the first implementation with a single return statement. Is too much easy to read and debug and also avoid situations like this one...that you don't really know what you should return to in this situation? 

So let's rollback to the first implementation, with a single return statement.